src/datajudge/constraints/base.py

```diff
@@ -149,7 +149,7 @@ def get_description(self) -> str:
 
     @property
     def target_prefix(self) -> str:
-        return f"{self.ref2.get_string()}'s " if (self.ref2 is not None) else ""
+        return f"{str(self.ref2)}'s " if (self.ref2 is not None) else ""
```

Comment:
It should be sufficient to do as follows
```suggestion
        return f"{self.ref2}'s " if (self.ref2 is not None) else ""
```

The `f-string` already calls the `__str__` method of the class.